common/roschannels/objstatechangechannel.hpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "roschannel.hpp"
+#include "atos_interfaces/msg/object_state_change.hpp"
+
+namespace ROSChannels {
+    namespace ObjectStateChange {
+        const std::string topicName = "object_state_change";
+        using message_type = atos_interfaces::msg::ObjectStateChange;
+        const rclcpp::QoS defaultQoS = rclcpp::QoS(rclcpp::KeepAll());
```

Comment:
A bit nitpicky perhaps but why is this called defaultQos? Can a node be expected to change its qos to keep(1) or something else? 

Also is this is more of a question, but is there a specific reason to use keep all instead of keep N? Does ROS handle this if it gets to large? I assume might not be that big of a risk considering that most runs do not include that many state changes.

Reply:
> A bit nitpicky perhaps but why is this called defaultQos?

The variable is named defaultQoS because it is the default QoS of the channel. 
> Can a node be expected to change its qos to keep(1) or something else?

When constructing a ROSChannels::ObjectStateChange::Pub or ::Sub object, another value for QoS, any value can be input as a constructor-parameter, but we have not implemented changing this value dynamically at runtime. 

>Also is this is more of a question, but is there a specific reason to use keep all instead of keep N?

This is the size of the callback queue for subscribers of this type. If there are system-wide delays there might be useful to have a complete history of state changes. Example with delays and KeepLast(2): off -> initialized -> disarmed -> armed -> disarmed. In this case, OBC will get a state change from off to disarmed which will be illegal. With the full history the state changes will be correct. I don't think that this is a major issue though.. 

>Does ROS handle this if it gets to large?

This is subject to the resources of the underlying DDS vendor (https://design.ros2.org/articles/qos.html).

>I assume might not be that big of a risk considering that most runs do not include that many state changes.
Yes I agree